Sources/CartonHelpers/DiagnosticsParser.swift

```diff
@@ -0,0 +1,207 @@
+// Copyright 2020 Carton contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import TSCBasic
+import Splash
+
+fileprivate extension StringProtocol {
+  func matches(regex: NSRegularExpression) -> String.SubSequence? {
+    let str = String(self)
+    let range = NSRange(location: 0, length: utf16.count)
+    guard let match = regex.firstMatch(in: str, options: [], range: range),
+          let matchRange = Range(match.range, in: String(self)) else {
+      return nil
+    }
+    return str[matchRange.upperBound..<str.endIndex]
+  }
+}
+
+fileprivate extension String.StringInterpolation {
+  mutating func appendInterpolation<T>(_ value: T, color: String...) {
+    appendInterpolation("\(color.map { "\u{001B}\($0)" }.joined())\(value)\u{001B}[0m")
+  }
+}
+
+fileprivate extension TokenType {
+  var color: String {
+    switch self {
+    case .keyword: return "[35;1m"
+    case .comment: return "[90m"
+    case .call, .dotAccess, .property, .type: return "[94m"
+    case .number, .preprocessing: return "[33m"
+    case .string: return "[91;1m"
+    default: return "[0m"
```

Comment:
Do these values match some specific colors? I also wonder how this looks in black text on white background terminals.

Would be great to provide a link to some docs that clarify how these values are inferred.

Reply:
Just tried with light mode, and found the ERROR text shows with black text on red, which I'll fix, but everything else looked good.

Theres's an article on Wikipedia about ANSI escape codes ([section on Colors](https://en.wikipedia.org/wiki/ANSI_escape_code#Colors)) I'll include a reference to it in the file.